src/models/graph.ts

```diff
@@ -0,0 +1,135 @@
+import { Json } from './json'
```

Comment:
this moves a lot of types that were originally defined in /utils/graph.ts, but improves upon typing as well